giza_actions/integrations/uniswap/utils.py

```diff
@@ -0,0 +1,88 @@
+import json
+import math
+
+from ape import Contract
+
+from .constants import MAX_TICK, MIN_TICK, Q96, TICKS_Q, _tick_spacing
+
+
+def load_json(path):
+    with open(path) as f:
+        return json.load(f)
+
+
+def load_contract(address):
+    return Contract(address)
+
+
+def price_to_tick(price):
+    sqrtPriceX96 = int(price * 2**96)
+    tick = math.floor(math.log((sqrtPriceX96 / Q96) ** 2) / math.log(TICKS_Q))
+    return tick
+
+
+def price_to_sqrtp(p):
+    return int(math.sqrt(p) * Q96)
+
+
+def tick_to_price(tick, decimals0, decimals1, invert=False):
+    if invert:
+        return 1 / (TICKS_Q**tick / 10 ** (decimals1 - decimals0))
+    else:
+        return TICKS_Q**tick / 10 ** (decimals1 - decimals0)
+
+
+def get_min_tick(fee):
+    min_tick_spacing = _tick_spacing[fee]
+    return -(MIN_TICK // -min_tick_spacing) * min_tick_spacing
+
+
+def get_max_tick(fee):
+    max_tick_spacing = _tick_spacing[fee]
+    return (MAX_TICK // max_tick_spacing) * max_tick_spacing
+
+
+def default_tick_range(fee):
+    min_tick = get_min_tick(fee)
+    max_tick = get_max_tick(fee)
+    return min_tick, max_tick
+
+
+# https://uniswapv3book.com/milestone_1/calculating-liquidity.html
+def calc_amount0(liq, pa, pb):
```

Comment:
Nitpick: I'm not really fond of using numbers in the function naming, whats the reasoning of differentiating `0` and `1`? Couldn't we add another thing besides de zero or one to explain the usage?

Reply:
It's the two tokens in a pool. They use token0 and token1 convention but I can change to tokenA and tokenB if you prefer